### PR TITLE
Assignment 2 Fixes

### LINKED ISSUE #81: Limit question title length else it will overflow

### PR DESCRIPTION
**Pull request by venuslimm**
This PR
- Removes Auth API (merged with User API)
- Updates env variables
- Updates `README.md`, `schema.sql`
- Updates `user/update` and `user/change-password` API endpoints
- Fixes question title overflow on /landing page

Closes #81

### CODE REVIEW

File: frontend/src/loadEnvironment.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
-const SERVER_HOST = process.env.REACT_APP_SERVER_HOST || 'http://localhost';
-const SERVER_PORT =  process.env.REACT_APP_SERVER_PORT || 5000;
+const SERVER_HOST = process.env.REACT_APP_HOST || 'http://localhost';
+const SERVER_PORT = process.env.SERVER_PORT || 3001;
```
[the remark this answers is not in the document]
Comment (venuslimm) [reply frontend/src/loadEnvironment.js:2]: Is this for all env variables or just the host and port numbers?

[the remark this answers is not in the document]
Comment (JrmCkh) [reply frontend/src/loadEnvironment.js:2]: All env variables

[the remark this answers is not in the document]
Comment (lhy-hoyin) [reply frontend/src/loadEnvironment.js:2]: Do check this out.

https://create-react-app.dev/docs/adding-custom-environment-variables/

[the remark this answers is not in the document]
Comment (venuslimm) [reply frontend/src/loadEnvironment.js:2]: Okay, will update all the env vars to start with REACT_APP. Thanks

**Review by JrmCkh — APPROVED** [review]
LGTM

**Review by cadencjk — APPROVED** [review]
LGTM

**Review by anthonynmh — APPROVED** [review]
LGTM

**Pull request merged**